packages/next/next-server/server/config.ts

```diff
@@ -89,6 +89,30 @@ function assignDefaults(userConfig: { [key: string]: any }) {
       )
     }
 
+    if (key === 'pageExtensions') {
+      const pageExtensions = userConfig[key]
+
+      if (!Array.isArray(pageExtensions)) {
```

Comment:
We should allow users to pass `pageExtensions: undefined`.

Let's allow that and add a test for it.